redfish-core/lib/log_services.hpp

```diff
@@ -4040,6 +4046,114 @@ inline void getSystemHardwareIsolationLogEntryCollection(
             "xyz.openbmc_project.HardwareIsolation.Create"});
 }
 
+/**
+ * @brief API Used to fill LogEntry schema by using the HardwareIsolation dbus
+ *        entry object which will get by using the given entry id in redfish
+ *        uri.
+ *
+ * @param[in] req - The HardwareIsolation redfish request (unused now).
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryId - The entry id of HardwareIsolation entries to retrieve
+ *                      the corresponding isolated hardware details.
+ *
+ * @return The redfish response in the given buffer with LogEntry schema
+ *         members if success else will error.
+ */
+inline void getSystemHardwareIsolationLogEntryById(
+    const crow::Request& /* req */,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& entryId)
+{
+    sdbusplus::message::object_path entryObjPath(
+        std::string(hwIsolationEntryRootDbusPath) + "/" + entryId);
+
+    auto getManagedObjectsRespHandler = [asyncResp, entryObjPath](
+                                            const boost::system::error_code ec,
+                                            const GetManagedObjectsType&
+                                                mgtObjs) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR
+                << "DBUS response error [" << ec.value() << " : "
+                << ec.message()
+                << "] when tried to get the HardwareIsolation managed objects";
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        bool entryIsPresent = false;
+        for (auto dbusObjIt = mgtObjs.begin(); dbusObjIt != mgtObjs.end();
+             dbusObjIt++)
+        {
+            if (dbusObjIt->first != entryObjPath)
```

Comment:
Its good if you use == operator and avoid using 'continue'. This comment applies to many other places in this PR